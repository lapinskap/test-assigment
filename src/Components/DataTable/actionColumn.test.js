import React from 'react';
import EnzymeToJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import ActionColumn from './actionColumn';
import '../../tests/setupTests';
import TestAppContext from '../../tests/TestAppContext';

describe('ButtonList', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const spyOnClick = jest.fn();
    const buttons = [
      {
        size: 'lg',
        color: 'success',
        className: 'mr-2',
        type: 'submit',
        disabled: false,
      },
      {
        size: 'lg',
        color: 'success',
        className: 'mr-2',
        text: 'Save',
        type: 'button',
        disabled: false,
      },
      {
        text: 'Delete',
        onClick: spyOnClick,
      },
      {
        size: 'sm',
        color: 'danger',
        className: 'random-classname',
        disabled: false,
        runBefore: null,
        href: '/',
      },

    ];
    const wrapper = mount((
      <TestAppContext router>
        <ActionColumn buttons={buttons} data={{ id: 1 }} />
      </TestAppContext>
    ));
    expect(EnzymeToJson(wrapper)).toMatchSnapshot();

    const firstButton = wrapper.find('Button').at(0);
    firstButton.simulate('click');

    const secondButton = wrapper.find('Button').at(1);
    secondButton.simulate('click');

    const thirdButton = wrapper.find('Button').at(2);
    thirdButton.simulate('click');

    expect(spyOnClick).toHaveBeenCalledTimes(1);
  });
});
