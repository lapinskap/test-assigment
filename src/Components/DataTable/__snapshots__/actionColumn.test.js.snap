// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonList renders correctly 1`] = `
<TestAppContext
  language={false}
  rbs={false}
  redux={false}
  router={true}
>
  <HashRouter
    getUserConfirmation={[Function]}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ActionColumn
        buttons={
          Array [
            Object {
              "className": "mr-2",
              "color": "success",
              "disabled": false,
              "size": "lg",
              "type": "submit",
            },
            Object {
              "className": "mr-2",
              "color": "success",
              "disabled": false,
              "size": "lg",
              "text": "Save",
              "type": "button",
            },
            Object {
              "onClick": [MockFunction],
              "text": "Delete",
            },
            Object {
              "className": "random-classname",
              "color": "danger",
              "disabled": false,
              "href": "/",
              "runBefore": null,
              "size": "sm",
            },
          ]
        }
        data={
          Object {
            "id": 1,
          }
        }
        rowId="id"
      >
        <RbsButton
          className="mr-2"
          color="success"
          data-t1="rowAction-undefined"
          data-t2={1}
          disabled={false}
          href={null}
          onClick={[Function]}
          permission=""
          size="lg"
          title="Edytuj"
          type="submit"
        >
          <Button
            className="mr-2"
            color="success"
            data-t1="rowAction-undefined"
            data-t2={1}
            disabled={false}
            href={null}
            onClick={[Function]}
            size="lg"
            tag="button"
            title="Edytuj"
            type="submit"
          >
            <button
              aria-label={null}
              className="mr-2 btn btn-success btn-lg"
              data-t1="rowAction-undefined"
              data-t2={1}
              disabled={false}
              href={null}
              onClick={[Function]}
              title="Edytuj"
              type="submit"
            >
              Edytuj
            </button>
          </Button>
        </RbsButton>
        <RbsButton
          className="mr-2"
          color="success"
          data-t1="rowAction-undefined"
          data-t2={1}
          disabled={false}
          href={null}
          onClick={[Function]}
          permission=""
          size="lg"
          title="Edytuj"
          type="button"
        >
          <Button
            className="mr-2"
            color="success"
            data-t1="rowAction-undefined"
            data-t2={1}
            disabled={false}
            href={null}
            onClick={[Function]}
            size="lg"
            tag="button"
            title="Edytuj"
            type="button"
          >
            <button
              aria-label={null}
              className="mr-2 btn btn-success btn-lg"
              data-t1="rowAction-undefined"
              data-t2={1}
              disabled={false}
              href={null}
              onClick={[Function]}
              title="Edytuj"
              type="button"
            >
              Edytuj
            </button>
          </Button>
        </RbsButton>
        <RbsButton
          className=""
          color="link"
          data-t1="rowAction-undefined"
          data-t2={1}
          disabled={false}
          href={null}
          onClick={[MockFunction]}
          permission=""
          size={null}
          title="Edytuj"
          type="button"
        >
          <Button
            className=""
            color="link"
            data-t1="rowAction-undefined"
            data-t2={1}
            disabled={false}
            href={null}
            onClick={[MockFunction]}
            size={null}
            tag="button"
            title="Edytuj"
            type="button"
          >
            <button
              aria-label={null}
              className="btn btn-link"
              data-t1="rowAction-undefined"
              data-t2={1}
              disabled={false}
              href={null}
              onClick={[Function]}
              title="Edytuj"
              type="button"
            >
              Edytuj
            </button>
          </Button>
        </RbsButton>
        <RbsButton
          className="random-classname"
          color="danger"
          data-t1="rowAction-undefined"
          data-t2={1}
          disabled={false}
          href="#/"
          onClick={[Function]}
          permission=""
          size="sm"
          title="Edytuj"
          type="button"
        >
          <Button
            className="random-classname"
            color="danger"
            data-t1="rowAction-undefined"
            data-t2={1}
            disabled={false}
            href="#/"
            onClick={[Function]}
            size="sm"
            tag="button"
            title="Edytuj"
            type="button"
          >
            <a
              aria-label={null}
              className="random-classname btn btn-danger btn-sm"
              data-t1="rowAction-undefined"
              data-t2={1}
              disabled={false}
              href="#/"
              onClick={[Function]}
              title="Edytuj"
              type="button"
            >
              Edytuj
            </a>
          </Button>
        </RbsButton>
      </ActionColumn>
    </Router>
  </HashRouter>
</TestAppContext>
`;
